js_modules/dagit/packages/core/src/instigation/Unloadable.test.tsx

```diff
@@ -1,140 +1,62 @@
-import {gql, useQuery} from '@apollo/client';
-import {act, render, screen} from '@testing-library/react';
+import {MockedProvider} from '@apollo/client/testing';
+import {render, screen} from '@testing-library/react';
 import * as React from 'react';
 
-import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
 import {InstigationStatus} from '../graphql/types';
-import {TestProvider} from '../testing/TestProvider';
 
-import {INSTIGATION_STATE_FRAGMENT} from './InstigationUtils';
-import {UnloadableSchedules, UnloadableSensors} from './Unloadable';
-import {UnloadableInstigationStatesQuery} from './types/Unloadable.test.types';
+import {UnloadableSensors} from './Unloadable';
+import {unloadableSchedule, unloadableSensor} from './__fixtures__/Unloadable.fixtures';
 
 describe('Unloadables', () => {
-  const UNLOADABLE_INSTIGATION_STATES_QUERY = gql`
-    query UnloadableInstigationStatesQuery {
-      unloadableInstigationStatesOrError {
-        ... on InstigationStates {
-          results {
-            id
-            ...InstigationStateFragment
-          }
-        }
-        ...PythonErrorFragment
-      }
-    }
-
-    ${INSTIGATION_STATE_FRAGMENT}
-    ${PYTHON_ERROR_FRAGMENT}
-  `;
-
-  const defaultMocks = {
-    InstigationStates: () => ({
-      results: () => [...new Array(1)],
-    }),
-  };
-
   describe('Sensors', () => {
-    const Test = () => {
-      const {data, loading} = useQuery<UnloadableInstigationStatesQuery>(
-        UNLOADABLE_INSTIGATION_STATES_QUERY,
-      );
-      if (loading) {
-        return <div>Loading…</div>;
-      }
-      const states = data?.unloadableInstigationStatesOrError;
-      if (states && states.__typename === 'InstigationStates') {
-        return <UnloadableSensors sensorStates={states.results} />;
-      }
-      return <div>Error!</div>;
-    };
-
     it('shows enabled ON switch if running and viewer has permission to edit sensors', async () => {
-      const mocks = {
-        InstigationState: () => ({
-          status: () => InstigationStatus.RUNNING,
-        }),
-      };
-
-      await act(async () => {
-        render(
-          <TestProvider
-            apolloProps={{mocks: [defaultMocks, mocks]}}
-            permissionOverrides={{edit_sensor: {enabled: true, disabledReason: null}}}
-          >
-            <Test />
-          </TestProvider>,
-        );
-      });
+      render(
+        <MockedProvider>
+          <UnloadableSensors sensorStates={[unloadableSensor]} />
+        </MockedProvider>,
+      );
```

Comment:
This is sooo much nicer

Reply:
This looks great! Does this UnloadableSensors component include a useQuery though? It seems odd that these tests all work without passing any mocks to the MockedProvider. I wonder if we could actually not wrap this in any Apollo provider at all?

Reply:
There are lower-level components within the tree that do require it, unfortunately.